extensions/default/src/DicomWebDataSource/utils/mergeUtils.js

```diff
@@ -0,0 +1,190 @@
+import dcmjs from 'dcmjs';
+
+import { retrieveStudyMetadata } from '../retrieveStudyMetadata';
+import mergeResults from './mergeResults';
+
+const { DicomMetaDictionary } = dcmjs.data;
+const { naturalizeDataset } = DicomMetaDictionary;
+
+const STUDY_INSTANCE_UID = '0020000D';
+const SERIES_INSTANCE_UID = '0020000E';
+
+/**
+ * Merges search results from multiple DICOMWeb clients.
+ * @param {Object} options - The options object.
+ * @param {Array} options.clients - The array of DICOMWeb clients to search.
+ * @param {Object} options.origParams - The original search parameters.
+ * @param {Function} options.mapParams - The function to map search parameters to each client.
+ * @param {Function} options.qidoSearch - The function to perform the QIDO search.
+ * @returns {Promise<Array>} - A promise that resolves to the merged search results.
+ */
+export const mergedSearch = async ({ clients, origParams, mapParams, qidoSearch }) => {
```

Comment:
If you truly need merge search results, then you need to define how the merge happens, and that should be in a separate data source type, NOT in the DICOMweb type.  That is, you might merge two DICOMweb data sources, and another one that is a different type entirely.  Mixing it up in DICOMweb code is going to cause endless bugs.

Reply:
It's not part of the plan to merge conflicting results. The assumption is that each server will contain a unique set of data. Real conciliation of data is much more complex and not part of the scope of the PR.